packages/theme/modules/checkout/composables/useCart/index.ts

```diff
@@ -197,21 +197,40 @@ PRODUCT
     }
   };
 
-  const applyCoupon = async ({ couponCode, customQuery }): Promise<void> => {
-    Logger.debug('useCart.applyCoupon');
+  const handleCoupon = async (couponCode = null, customQuery = null): Promise<boolean> => {
```

Comment:
```suggestion
  const handleCoupon = async (couponCode = null, customQuery = null): Promise<void> => {
```
What do you think about throwing an exception from handleCoupon if there are errors instead of returning false? You could then handle the exception with try/catch in applyCoupon

To me, a function succeeded if executed fully without throwing an excception, not by returning `true`

Reply:
ok